src/main.cpp

```diff
@@ -183,6 +183,58 @@ int main(int argc, char **argv)
 
 	QGuiApplication::setDesktopSettingsAware(false);
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+	////////////////////////////////////////////////////////////////////////////////////////////////
+	/// We can define our default OpenGL context only with Qt6 when we have no more dynamic OpenGL/ANGLE.
+	/// IMPORTANT: OpenGL default context/formats must be configured before constructing app!
+	/// Copy from StelMainView
+	/// TODO: adapt/remove code from StelMainView?
+	/// TODO: Find out of this works on GLES devices!
+
+	//use the default format as basis
+	QSurfaceFormat fmt = QSurfaceFormat::defaultFormat();
+
+	//if on an GLES build, do not set the format
+	//if (fmt.renderableType()==QSurfaceFormat::OpenGL) // Probably at this point it is only QSurfaceFormat::DefaultRenderableType
+	{
+#ifdef Q_OS_MACOS
+		// On OSX, you should get what you ask for. Setting format later may not work.
+		// Let's assume all MacOSX deliver at least 3.3 compatibility profile.
+		fmt.setMajorVersion(3);
+		fmt.setMinorVersion(3);
+		fmt.setProfile(QSurfaceFormat::CompatibilityProfile);
```

Comment:
I don't know if Mac allows this. Maybe they only have 3.3 Core profile. However we require some compatibility profile. Or Mac should really only provide a 2.1 context.